web/packages/teleport/src/TopBar/TopBarSideNav.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import styled, { useTheme } from 'styled-components';
+import { Link } from 'react-router-dom';
+import { Flex, Image, TopNav } from 'design';
+import { matchPath, useHistory } from 'react-router';
+import { Theme } from 'design/theme/themes/types';
+import { HoverTooltip } from 'shared/components/ToolTip';
+
+import useTeleport from 'teleport/useTeleport';
+import { UserMenuNav } from 'teleport/components/UserMenuNav';
+import { useFeatures } from 'teleport/FeaturesContext';
+import cfg from 'teleport/config';
+import { useLayout } from 'teleport/Main/LayoutContext';
+import { logos } from 'teleport/components/LogoHero/LogoHero';
+
+import { Notifications } from 'teleport/Notifications';
+
+export function TopBar({ CustomLogo }: TopBarProps) {
+  const ctx = useTeleport();
+  const history = useHistory();
+  const features = useFeatures();
+  const { currentWidth } = useLayout();
+  const theme: Theme = useTheme();
+
+  // find active feature
+  const feature = features
+    .filter(feature => Boolean(feature.route))
+    .find(f =>
+      matchPath(history.location.pathname, {
+        path: f.route.path,
+        exact: f.route.exact ?? false,
+      })
+    );
+
+  history?.location?.pathname === cfg.routes.downloadCenter;
+  const iconSize =
+    currentWidth >= theme.breakpoints.medium
+      ? navigationIconSizeMedium
+      : navigationIconSizeSmall;
+
+  return (
+    <TopBarContainer navigationHidden={feature?.hideNavigation}>
+      {!feature?.hideNavigation && <TeleportLogo CustomLogo={CustomLogo} />}
+      {!feature?.logoOnlyTopbar && (
+        <Flex height="100%" alignItems="center">
+          <Notifications iconSize={iconSize} />
+          <UserMenuNav username={ctx.storeUser.state.username} />
+        </Flex>
+      )}
+    </TopBarContainer>
+  );
+}
+
+export const TopBarContainer = styled(TopNav)`
+  position: absolute;
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  background: ${p => p.theme.colors.levels.surface};
+  overflow-y: initial;
+  overflow-x: none;
+  flex-shrink: 0;
+  z-index: 23;
```

Comment:
This would be good to extract too if this value is supposed to be kept in sync with other side nav z-index values. If you don't want to end up with TopBar importing the whole `SideNavigation/Navigation.tsx`, I'd say having a separate file with the z-index map is not a bad idea.